src/components/sectionList/modelValue/PublicAccess.tsx

```diff
@@ -0,0 +1,29 @@
+import i18n from '@dhis2/d2-i18n'
+import React from 'react'
+import { Sharing } from '../../../types/generated'
+
+export const isSharing = (value: unknown): value is Sharing => {
+    return typeof (value as Sharing).public === 'string'
+}
+
+const getPublicAccessString = (value: Sharing): string => {
+    const publicAccessString = value.public
+    const metadata = publicAccessString.substring(0, 2)
+    const data = publicAccessString.substring(2, 4)
+    const other = publicAccessString.substring(4)
+
+    if (other === '----' && (data === '--' || data === 'r-' || data === 'rw')) {
+        if (metadata === 'rw') {
+            return i18n.t('Public can edit')
+        } else if (metadata === 'r-') {
+            return i18n.t('Public can view')
+        } else if (metadata === '--') {
+            return i18n.t('Public can not access')
+        }
+    }
```

Comment:
This logic is mostly taken from old `d2-ui`. Seems a bit weird, but didn't want to refactor much as to not introduce bugs.

Reply:
I agree that this looks weird 😅, but I understood it (eventually)

Reply:
I'm not sure why it needs to check the values of `other` and `data`?

Reply:
I assume it's just doing that to check that it's in the right format? E.g. instead of testing against a RegEx (like `^(r-|rw|--){2}-{4}$`

Reply:
If we were to refactor it, why would we be so verbose? I guess it could be more readable, but the multiple tests seem redundant to me. When you could do it in the same regex, like the one Tom provided?

```ts
  const publicAccessRegex = /^(r-|rw|--){2}(-){4}$/;
  
  const matches = value.match(publicAccessRegex);
  if(!matches) {
    return 'N/A'
  }
  const [_, metadata] = matches

  if (metadata === "rw") {
    return i18n.t("Public can edit");
  } else if (metadata === "r-") {
    return i18n.t("Public can view");
  }

  return i18n.t("Public cannot access");
```

Reply:
> If we were to refactor it, why would we be so verbose? I guess it could be more readable, but the multiple tests seem redundant to me. When you could do it in the same regex, like the one Tom provided?

It explains the different cases without having to explain it with comments or tests. I guess this boils down to personal style, but I favor clarity over dry- & compact-ness. I rather spend more time reading than spending more time figuring out how and why something works in a specific way. IMO clarity saves more time, even if it introduces some redundancy. With clarity I mean that the code includes the reason why a certain input is invalid in constrast to just knowing that it's not valid. When using just the regex, I wouldn't understand why we check the non-public access rights. (Although I think my actual preferred solution would be to separate these two completely (checking for validity and checking for access, as these are two different things done by one function with a name that doesn't indicate the validity check whatsoever. But I lack context, so I have no idea why it's been implemented the way it's been implemented)

Not saying that my solution is perfect though, just what I would've done initially

Reply:
> Although I think my actual preferred solution would be to separate these two completely 

Yeah, I've been thinking having something like `parsePublicAccessString`. That would check the validity and return a parsed object `metadata: { read: boolean, write: boolean } `.
So consumers would deal with the parsed object instead.

Reply:
That sounds a lot more reasonable to me as well

Reply:
Implemented this now, with test-cases that should basically cover any valid combination at least.
https://github.com/dhis2/maintenance-app-beta/pull/350/files#diff-c8b030c40f877769002867a17865c8ad03c613a132a648b280de2a8bd919e55d

Reply:
That's a lot better! Now that I see it implemented, I wonder how much value `N/A` adds.. Maybe a question for Joe, but I guess it'd be better to also display "Public cannot access" as the user doesn't know what `N/A` means.
Just food for thought: Maybe we should throw an error in `N/A` cases and display some kind of warning/error (alert bar?) telling the user that there's an issue with the sharing settings

Reply:
> I wonder how much value N/A adds.. 

Yeah, I've had the same thought. Maybe we could even just render a blank value?

> Just food for thought: Maybe we should throw an error in N/A cases and display some kind of warning/error

Yeah I have thought about having an  `errorBoundary` in the `ModelValue`-component, that could catch errors render a unified error-component. We wouldn't want to crash the whole list just because one column failed. Can talk to Joe, since there's limited space to describe the error in a column. 
Don't think an alert-bar makes much sense though, since multiple error could happen across the rendered columns.

Reply:
> Don't think an alert-bar makes much sense though, since multiple error could happen across the rendered columns.

:+1: 

> Can talk to Joe, since there's limited space to describe the error in a column.

:ok_hand:

Reply:
> perhaps just print Error  in the cell with some special styling (I'll think of something), then console log the rest of the error details for someone doing the debugging 

Will just change to error, and we can apply styles later.